stacks_queues/queue.py

```diff
@@ -1,61 +1,91 @@
 
 INITIAL_QUEUE_SIZE = 20
 
+
 class QueueFullException(Exception):
     pass
 
+
 class QueueEmptyException(Exception):
     pass
 
+
 class Queue:
 
     def __init__(self):
         self.store = [None] * INITIAL_QUEUE_SIZE
         self.buffer_size = INITIAL_QUEUE_SIZE
-        self.front = -1
+        self.front = -1  # to show there is nothing in the queue yet
         self.rear = -1
         self.size = 0
-      
 
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+        if self.front == -1:
+            self.front = 0
+            self.rear = 0
+        elif self.size == self.buffer_size:
+            raise QueueFullException('Queue is full')
+
+        self.store[self.rear] = element
+        self.rear = (self.rear + 1) % self.buffer_size
+        self.size += 1
 
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
         """
-        pass
+        if self.size == 0:
+            raise QueueEmptyException('Queue is empty')
+
+        head = self.front
+        element = self.store[head]
+
+        for i in range(self.size):
+            self.store[i] = self.store[i+1]
 
-    def front(self):
+        self.rear -= 1
+        self.size -= 1
+
+        return element
+
+    def get_front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
-        pass
-        
+        return self.store[self.front] if self.size > 0 else None
 
     def size(self):
         """ Returns the number of elements in
             The Queue
         """
-        pass
+        return self.size
 
     def empty(self):
         """ Returns True if the Queue is empty
             And False otherwise.
         """
-        pass
+        return self.size == 0
 
     def __str__(self):
         """ Returns the Queue in String form like:
             [3, 4, 7]
             Starting with the front of the Queue and
             ending with the rear of the Queue.
         """
-        pass
+        result = []
+
+        if self.front == -1:
+            return "[]"
+
+        while self.front < self.size:
```

Comment:
WIth the suggestion to `dequeue` above, front might no longer be at 0, and the buffer might wrap around, so we'd need a different condition and update to front.